core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/constraints/SpeedConstraint.java

```diff
@@ -20,8 +19,11 @@ public SpeedConstraint(double speedConstraint,
     public boolean initCheck(double direction, double position, double speed) {
         if (type == CEILING)
             return speed <= speedConstraint;
-        else
+        if (type == FLOOR)
             return speed >= speedConstraint;
+        if (type == MAINTAIN_SPEED)
+            return speed == speedConstraint;
+        return true;   // default return
```

Comment:
With a switch, there's no need for a default return.

(This "please use a switch" comment also applies to the rest of the `if` chains with enums)